tests/test_checkpoint.rs

```diff
@@ -0,0 +1,141 @@
+// Copyright 2018 Eugene P.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+extern crate rocksdb;
+
+use rocksdb::{checkpoint::Checkpoint, DB, Options};
+
+fn remove_dir_silently(path: &String) {
+    use std::fs::remove_dir_all;
+
+    match remove_dir_all(&path) {
+        Ok(_) => {}
+        Err(_) => {}
+    };
+}
+
+#[test]
+pub fn test_single_checkpoint() {
+    const PATH_PREFIX: &str = "_rust_rocksdb_cp_single_";
+
+    // Create DB with some data and make a checkpoint
+    let cp1;
+    let cp1_path;
+    let db_path;
+    {
+        db_path = PATH_PREFIX.to_string() + "db1";
+
+        remove_dir_silently(&db_path);
+
+        let mut opts = Options::default();
+        opts.create_if_missing(true);
+        let db = DB::open(&opts, db_path.clone()).unwrap();
+
+        db.put(b"k1", b"v1").unwrap();
+        db.put(b"k2", b"v2").unwrap();
+        db.put(b"k3", b"v3").unwrap();
+        db.put(b"k4", b"v4").unwrap();
+
+        cp1 = Checkpoint::new(&db).unwrap();
+        cp1_path = PATH_PREFIX.to_string() + "cp1";
+        cp1.create_checkpoint(cp1_path.clone()).unwrap();
+    }
+
+    // Create a DB from the checkpoint
+    {
+        let db = DB::open_default(cp1_path.clone()).unwrap();
+
+        assert_eq!(*db.get(b"k1").unwrap().unwrap(), *b"v1");
+        assert_eq!(*db.get(b"k2").unwrap().unwrap(), *b"v2");
+        assert_eq!(*db.get(b"k3").unwrap().unwrap(), *b"v3");
+        assert_eq!(*db.get(b"k4").unwrap().unwrap(), *b"v4");
+    }
+
+    remove_dir_silently(&db_path);
+    remove_dir_silently(&cp1_path);
+}
+
+#[test]
+pub fn test_multi_checkpoints() {
+    const PATH_PREFIX: &str = "_rust_rocksdb_cp_multi_";
+
+    // Create DB with some data and make a checkpoint
+    let cp1;
+    let cp1_path;
+    let db_path;
+    {
+        db_path = PATH_PREFIX.to_string() + "db1";
```

Comment:
I suppose `format!` marco is more suitable in this case.
`let db_path = format!("{}db1", PATH_PREFIX);`